app.ts

```diff
@@ -0,0 +1,211 @@
+/** @format */
+
+interface IUser {
+    firstName: string;
+    secondName: string;
+    role: string;
+    age: number;
+    gender: string;
+    isAdmin?: boolean;
+    specialization?: string;
+    course?: number;
+    faculty?: string;
+}
+
+class DataBase implements DataAggregator {
+    private static instance: DataBase;
+    private users: IUser = [];
+
+    public static getInstance(): DataBase {
+        if (!DataBase.instance) {
+            DataBase.instance = new DataBase();
+        }
+
+        return DataBase.instance;
+    }
+
+    public addUser(user: IUser): void {
+        this.users.push(user);
+    }
+
+    public getUsers(): IUser {
+        return this.users;
+    }
+
+    public removeUser(value: string, key: string): IUser {
+        return this.users.filter((item: IUser) => item[key] !== value)
+    }
+
+    public createIterator(): DataBaseIterator {
+        return new DataBaseIterator(this)
+    }
+}
+
+interface DataAggregator {
+    getUsers(): object;
+}
+
+class DataBaseIterator {
+    private position: number = 0;
+
+    constructor(private users: DataAggregator) { }
+
+    public current(index: number = this.position) {
+        this.position = index;
+        return this.users.getUsers()[index];
+    }
+
+    public dataLenght() {
+        return this.users.getUsers()
```

Comment:
The return type of this method is confusing. It doesn't return Length of the Data, it returns an array of users